gui/src/2D/controls/inputTextArea.ts

```diff
@@ -0,0 +1,1282 @@
+
+                    let width = context.measureText(line.text.substring(highlightCursorLeft, highlightCursorRight)).width;
+                    let leftOffsetWidth = context.measureText(line.text.substring(0, highlightCursorLeft)).width + xOffset;
+
+                    this._highlightedText += line.text.substring(highlightCursorLeft, highlightCursorRight);
+                    
+                    context.globalAlpha = this._highligherOpacity;
+                    context.fillStyle = "green"; // this._textHighlightColor;
+                    
+                    context.fillRect(leftOffsetWidth, yOffset + currentIndex * this._fontOffset.height, width, this._fontOffset.height);
+                    context.globalAlpha = 1.0;
+                });
+            }
+            
+            context.restore();
+
+            // Render cursor
+            if (!this._blinkIsEven) {
+                //Cursoroffset text is counted from left to right increasing
+                const cursorOffsetText = this._lines[this._selectedLineIndex].text.substr(this._lines[this._selectedLineIndex].text.length - this._cursorOffset);
+                const cursorOffsetWidth = context.measureText(cursorOffsetText).width;
+
+                let cursorLeft = this._scrollLeft + this._lines[this._selectedLineIndex].width - cursorOffsetWidth;
+
+                if (cursorLeft < clipTextLeft) {
+                    this._scrollLeft += (clipTextLeft - cursorLeft);
+                    cursorLeft = clipTextLeft;
+                    this._markAsDirty();
+                } else if (cursorLeft > clipTextLeft + this._availableWidth) {
+                    this._scrollLeft += (clipTextLeft + this._availableWidth - cursorLeft);
+                    cursorLeft = clipTextLeft + this._availableWidth;
+                    this._markAsDirty();
+                }
+
+                let cursorTop = this._currentMeasure.top + this._margin.getValueInPixel(this._host, this._tempParentMeasure.height); //cursorTop distance from top to cursor start
+                if (this.lastClickedCoordinateY <= this._fontOffset.height) {
+                    // Nothing to do here
+                } else if (this.lastClickedCoordinateY > this._fontOffset.height * (this._lines.length - 1)) {
+                    cursorTop += (this._fontOffset.height * (this._lines.length - 1));
+                } else {
+                    cursorTop += (Math.floor(this.lastClickedCoordinateY / this._fontOffset.height)) * this._fontOffset.height;
+                }
+
+                context.fillRect(cursorLeft, cursorTop, 2, this._fontOffset.height);
+            }
+        }
+        context.restore();
+
+        clearTimeout(this._blinkTimeout);
+        this._blinkTimeout = <any>setTimeout(() => {
+            this._blinkIsEven = !this._blinkIsEven;
+            this._markAsDirty();
+        }, 500);
+        context.restore();
```

Comment:
I understand why this is needed, but this could be solved within the render loop using timers - 
https://doc.babylonjs.com/divingDeeper/events/observables#observable-based-countdown-function
setTimeout could cause unexpected results, as it will surely run outside of the render loop

Reply:
Oh, great!

Fneitzel uses the same method as in `inputText.ts`, but I can improve it today.

Reply:
Currently, I'm facing some issues over AdvancedTimer class. 
- Is it correct to instantiate it in `_link` function to have access to the scene ?
- I can't stop/start in the same loop to reset the timer. 

I want to avoid SetAndStart to save some perf but it becomes to be the only solution...
Have you some ideas to bypass this ?

Reply:
I can't restart the timer during the same frame which it's stop